integr8ly/grafana-operator
# add operatorhub bundle files

Add operatorhub bundle files for the 1.3.0 release

Verification steps:

1. An OpenShift 4 cluster is needed
2. Login to your OpenShift 4 cluster with `oc` and create the following `OperatorSource`:

```yaml
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: pb82-operators
  namespace: openshift-marketplace
spec:
  type: appregistry
  endpoint: https://quay.io/cnr
  registryNamespace: pb82
  displayName: "Integreatly Operators"
  publisher: "Integreatly"
```

```sh
oc create -f ./operator-source.yaml
```

3. Make sure the operator source is valid:

```sh
oc get operatorsource -n openshift-marketplace
```

There should be a value like this:

```
pb82-operators        appregistry   https://quay.io/cnr   pb82                  Integreatly Operators   Integreatly   Succeeded   The object has been successfully reconciled   22m
```

4. In the OpenShift console, create a new project named `grafana`
5. In the OpenShift console, navigate to `Catalog -> OperatorHub`
6. In the left panel under `Provider Type` there should be a category `Custom`
7. It might take a minute for OpenShift to recognize the new OperatorSource, refresh the page if it's not yet visible.
8. Under the `Custom` category, you should find the Grafana Operator.
9. Click on it, then select `Install` (make sure that the Grafana project is selected)
10. Click on the `Suscribe` button and wait for the installation to finish.
11. Make sure the operator is running in the grafana namespace and does not have any errors in the log.
12. Create a Grafana Instance: `Grafana.yaml` and `GrafanaWithRoute.yaml` from the examples directory will both work, `GrafanaWithIngressHost.yaml` will only work on Kubernetes.
13. Create dashboards and a data source and make sure everything works as expected.
14. Log into the Grafana instance (credentials can be found in the CR) and make sure that the dashboards and data source have been created.
